### PR TITLE
Tagged transaction queue integration

### PR DESCRIPTION
**Pull request by tomusdrw**
Improvements (TODOs in the code; subsequent PRs):
- [ ] Correct longevity handling (currently it's time-based, not block-based)
- [ ] Pool based next-index calculation in consensus for misbehaviour reporting
- [ ] Parallel queue reads
- [ ] Optimize tags calculation (re-use provided tags from the queue)
- [x] Solve the annoying `Into<u64>` stuff for BlockNumber
- [ ] Handle transaction queue options (limits)
- [ ] More tests (especially listener stuff)
- [ ] Runtime: Change `TransactionValidity` to have named fields (revert `01a7d1a`)

### CODE REVIEW

File: core/primitives/src/hasher.rs [hunk 1]
```diff
@@ -42,6 +42,13 @@ pub mod blake2 {
 	#[derive(Debug)]
 	pub struct Blake2Hasher;
 
+	impl Blake2Hasher {
```
Comment (gavofyork) [inline core/primitives/src/hasher.rs]: What's wrong with `use Hasher`?

**Comment by gavofyork** [comment]
doesn't build (and needs resolving...)

File: core/primitives/src/hasher.rs (hunk 1 above)
Comment (tomusdrw) [reply core/primitives/src/hasher.rs]: It's only in `hash_db` crate - this was a temporary, quick and dirty way to move forward. I now changed that and `Hasher` trait is re-imported from `primitives`, do you feel it's ok?

File: core/transaction-pool/graph/src/pool.rs [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+		},
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	#[test]
+	#[ignore]
+	fn should_have_some_basic_tests() {
+		assert_eq!(true, false);
```
[the remark this answers is not in the document]
Comment (tomusdrw) [reply core/transaction-pool/graph/src/pool.rs]: We already have unit tests for `base_pool` and more integration-level tests in `transaction-pool/src/tests.rs`. This should mostly cover the listener stuff that I mentioned in the description.

**Pull request merged**